docs/esa/hubble.rst

```diff
@@ -89,8 +143,147 @@ It will also download a file storing all metadata for all observations
+  INFO: select o.*, p.calibration_level, p.data_product_type from ehst.observation 
+  AS o LEFT JOIN ehst.plane as p on o.observation_uuid=p.observation_uuid
+  where(p.calibration_level LIKE '%PRODUCT%' AND p.data_product_type LIKE '%image%'
+  AND o.intent LIKE '%SCIENCE%' AND (o.collection LIKE '%HLA%') AND (o.instrument_name
+  LIKE '%WFC3%') AND (o.instrument_configuration LIKE '%F555W%' OR
+  o.instrument_configuration LIKE '%F606W%')) [astroquery.esa.hubble.core]
+  >>> str(result)
+  Table length=2000
+  ...
+ 
+This will make a synchronous search, limited to 2000 results to find the observations that match these specific
+requirements. It will also download a votable file called **output.vot.gz** containing the result of the
+query.
+
+The following example uses the string definition of the calibration level ('PRODUCT') and executes an asynchronous job to get all the results that match the criteria.
+
+.. code-block:: python
+
+  >>> from astroquery.esa.hubble import ESAHubble
+  >>> esahubble = ESAHubble()
+  >>> result = esahubble.query_by_criteria(calibration_level = 'PRODUCT', 
+                                           data_product_type = 'image', 
+                                           intent='SCIENCE', 
+                                           obs_collection=['HLA'], 
+                                           instrument_name = ['WFC3'], 
+                                           filters = ['F555W', 'F606W'], 
+                                           async_job = True, 
+                                           output_file = 'output2.vot.gz', 
+                                           output_format="votable", 
+                                           verbose = True, 
+                                           get_query = False)
+  >>> str(result)
+  Table length=12965
+  ...
+
+As it has been mentioned, these parameters are optional and it is not necessary to define all of them to execute this function, as the search will be adapted to the ones that are available.
+
+.. code-block:: python
+
+  >>> from astroquery.esa.hubble import ESAHubble
+  >>> esahubble = ESAHubble()
+  >>> result1 = esahubble.query_by_criteria(calibration_level = 'PRODUCT', 
+                                           async_job = False, 
+                                           output_file = 'output3.vot.gz')
+  >>> result2 = esahubble.query_by_criteria(data_product_type = 'image', 
+                                            intent='SCIENCE', 
+                                            async_job = False, 
+                                            output_file = 'output4.vot.gz')
+  >>> result3 = esahubble.query_by_criteria(data_product_type = 'timeseries',
+                                            async_job = False, 
+                                            output_file = 'output5.vot.gz')
+ 
+If none of them are defined, this function will retrieve all the observations.
```

Comment:
```suggestion
If no criteria are specified to limit the selection, this function will retrieve all the observations.
```

Does "all the observations" mean *every* observation in the ESA-Hubble archive?  That sounds like it might take a while!

Reply:
We also wanted to cover the "no criteria" case to avoid failures in the algorithm in case no criteria is defined. If it is executed synchronously, it will only retrieve the first 2000 results. If it is executed asynchronously... yes, it might take a while!